src/Confluent.Kafka/Consumer.cs

```diff
@@ -266,6 +266,38 @@ IAsyncDeserializer<TValue> taskValueDeserializer
             => (keyDeserializer != null && valueDeserializer != null)
                 ? Consume<TKey, TValue>(timeout.TotalMillisecondsAsInt(), keyDeserializer, valueDeserializer) // fast path for simple case
                 : Consume(timeout.TotalMillisecondsAsInt());
+
+
+        /// <summary>
+        ///     Commits an offset based on the topic/partition/offset of a ConsumeResult.
+        /// </summary>
+        /// <param name="result">
+        ///     The ConsumeResult instance used to determine the committed offset.
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     A cancellation token that can be used to cancel this operation
+        ///     (currently ignored).
+        /// </param>
+        /// <exception cref="Confluent.Kafka.KafkaException">
+        ///     Thrown if the request failed.
+        /// </exception>
+        /// <exception cref="Confluent.Kafka.TopicPartitionOffsetException">
+        ///     Thrown if the result is in error.
+        /// </exception>
+        /// <remarks>
+        ///     A consumer which has position N has consumed messages with offsets up to N-1 
+        ///     and will next receive the message with offset N. Hence, this method commits an 
+        ///     offset of <paramref name="result" />.Offset + 1.
+        /// </remarks>
```

Comment:
There is no indication, apart from the unused cancellation token,  if this is a synchronous/blocking call or not

Reply:
This is idiomatic, pre `async`/`await`. e.g. `System.IO.File.ReadAllLines`. Ideally I'd implement this only as an async method, with cancellation token, and this would be clear. I really don't like that there's no distinction as well. It will become clear though when Async variants are implemented, because the blocking functions will all have Async equivalents. The API will be analogous to `System.IO.File` (and others).

Noteworthy: in the next unopened PR, which brings consistency to all the cancellation tokens / timeout parameters across the library, and i think the best thing to do right now is remove all the unused cancellation tokens (and consider adding them, with defaults, a non-breaking change).

Reply:
I don't think it hurts with a small doc string for the blocking behaviour.